picard/coverart/processing/processors.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright (C) 2024 Giorgio Fontanive
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+import time
+
+from PyQt6.QtCore import (
+    QBuffer,
+    Qt,
+)
+from PyQt6.QtGui import QImage
+
+from picard import log
+from picard.config import get_config
+from picard.extension_points.cover_art_processors import (
+    ImageProcessor,
+    ProcessingTarget,
+    register_cover_art_processor,
+)
+
+
+def _get_image_data(image, image_format):
+    buffer = QBuffer()
+    image.save(buffer, image_format, quality=90)
```

Comment:
If I understand correctly: when the `image.save` function returns false, we should still set whatever data we already have to the `coverartimage`. Then we could append an error to the `Album`, that way it's clearer to the user that the processing did not run correctly, instead of just writing to the log.